src/client/components/ProcessesListing.tsx

```diff
@@ -13,23 +13,28 @@ type ProcessProps = {
 
 const Process: React.FunctionComponent<ProcessProps> = (props: ProcessProps) => {
     const thread = props.process.threads.length && props.process.threads[0];
-    const statusText = thread
-        ? thread.status
-            + (thread.status === 'stopped' && thread.stoppedAt ? ` at line ${thread.stoppedAt}` : '')
-        : null;
+    let statusText: string = null;
+    if (thread) {
+        statusText = thread.status
+            + (thread.status === 'stopped' && thread.stoppedAt ? ` at line ${thread.stoppedAt}` : '');
+    } else if (props.process.runState === 'Z') {
+        // "zombie" / unreaped is a processwide, not per-thread, state
+        // in fact, when in this state, we won't have threads
+        // create the friendly status here
+        statusText = 'zombie';
+    }
     return (
         <div className="process">
             <div className="process-header">
                 <Pill text={`pid ${props.process.pid}`} color={props.color} />
                 {props.process.command}
-                {thread && ` (${statusText})`}
-                {props.process.threads.length
-                    && (
-                        <DebugControls
-                            debugServer={props.debugServer}
-                            thread={props.process.threads[0]}
-                        />
-                    )}
+                {statusText && ` (${statusText})`}
+                {(props.process.threads[0] && (
```

Comment:
If I recall correctly, if this is posed as `props.process.threads.length && ...`, then when threads.length is zero, the expression collapses to zero since zero is falsey, so it takes the left-hand of the and. The problem with this is zero actually renders into the DOM, which is not desired.